trussworks/react-uswds
# ci: Fix build not including typescript types

# Summary

Looks like when the TS version was bumped, the refactor to remove `awesome-typescript-loader` and rely purely on babel didn't backfilll for type generation which babel does not do on its own. In this PR we manually call `tsc` in the build step and let it emit the type declarations. This is the recommended approach for projects using babel alone and not relying on ts-loader.

Alternative considered: use ts loader on ts|x files before running babel. ts-loader is a bit slow, and felt like a more repetitive solution, using two loaders on all of our ts/tsx files when `tsc` will directly create just the type exports for us without any additional config.

Resources: 
- https://www.typescriptlang.org/docs/handbook/babel-with-typescript.html

## Related Issues or PRs

closes: https://github.com/trussworks/react-uswds/issues/2053
stems from: https://github.com/trussworks/react-uswds/pull/1548

## How To Test

Checkout the code, clean your local files (delete node_modules and lib), freshly install packages with yarn. running `yarn` will also run `yarn build` which will create the lib directory where you can confirm that .d.ts files are now output correctly again. Running `yarn pack` will generate the tarball which you can also verify includes the types.

```
> git checkout bl-fix-types
> rm -rf node_modules
> rm -rf lib
> yarn
> yarn pack
```

### Screenshots (optional):
<img width="224" alt="image" src="https://user-images.githubusercontent.com/15805554/167488257-7e00bc73-3f9c-443f-b2e7-fb027ec8f967.png">
